src/info/infoBackup.c

```diff
@@ -311,6 +313,64 @@ infoBackupData(const InfoBackup *this, unsigned int backupDataIdx)
     FUNCTION_LOG_RETURN(INFO_BACKUP_DATA, *((InfoBackupData *)lstGet(this->backup, backupDataIdx)));
 }
 
+/***********************************************************************************************************************************
+Delete a backup from the current backup list
+***********************************************************************************************************************************/
+void
+infoBackupDataDelete(const InfoBackup *this, const String *backupDeleteLabel)
+{
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(INFO_BACKUP, this);
+        FUNCTION_LOG_PARAM(STRING, backupDeleteLabel);
+    FUNCTION_LOG_END();
+
+    ASSERT(this != NULL);
+
+    for (unsigned int idx = 0; idx < infoBackupDataTotal(this); idx++)
+    {
+        InfoBackupData backupData = infoBackupData(this, idx);
+        if (strCmp(backupData.backupLabel, backupDeleteLabel) == 0)
+        {
+            lstRemove(this->backup, idx);
+        }
+    }
+
+    FUNCTION_LOG_RETURN_VOID();
+}
+
+/***********************************************************************************************************************************
+Return a list of current backup labels, applying a regex filter if provided
+***********************************************************************************************************************************/
+StringList *
+infoBackupDataLabelList(const InfoBackup *this, InfoBackupDataLabelListParam param)
+{
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(INFO_BACKUP, this);
+        FUNCTION_LOG_PARAM(STRING, param.filter);
+    FUNCTION_LOG_END();
+
+    ASSERT(this != NULL);
+
+    // Return a 0 sized list if no current backups or none matching the filter
+    StringList *result = strLstNew();
+
+    MEM_CONTEXT_TEMP_BEGIN()
+    {
+        // For each backup label, compare it to the filter (if any) and sort it for return
+        for (unsigned int backupLabelIdx = 0; backupLabelIdx < infoBackupDataTotal(this); backupLabelIdx++)
+        {
+            InfoBackupData backupData = infoBackupData(this, backupLabelIdx);
+            if (param.filter == NULL || regExpMatchOne(param.filter, backupData.backupLabel))
```

Comment:
It's a bit wasteful to use regExpMatchOne here.  Parsing regexps is relatively expensive.

Create the RegExp in advance and use it multiple times like storagePosixList does.

Reply:
done